src/pkg/terminator/reconciler.go

```diff
@@ -132,9 +133,15 @@ func (r Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (recon
 		evt := r.Parse(ctx, msg)
 		ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("event", evt))
 
+		evtAction := actionForEvent(evt, terminator)
+		ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("action", evtAction))
+
 		allInstancesHandled := true
+		ec2InstanceIDs := evt.EC2InstanceIDs()
 		savedCtx := ctx
-		for _, ec2InstanceID := range evt.EC2InstanceIDs() {
+		for i := 0; i < len(ec2InstanceIDs) && evtAction != v1alpha1.Actions.NoAction; i++ {
```

Comment:
The condition on this for loop is a little hard to parse. I think the intent is "if `evtAction` is `NoAction`, skip the entire for loop". If that's the case, it would be easier to read if that check were in an explicit `if` statement before the loop starts (but that does increase the nesting on this already heavily nested portion of code). Alternatively, perhaps this is an opportunity to factor out some of the logic into a separate function.

I don't have a strong opinion on this, but I wanted to hear your reasoning behind this choice.